lib/ansible/plugins/inventory/aws_ec2.py

```diff
@@ -48,8 +48,10 @@
               - name: AWS_SESSION_TOKEN
               - name: EC2_SECURITY_TOKEN
         regions:
-          description: A list of regions in which to describe EC2 instances. By default this is all regions except us-gov-west-1
-              and cn-north-1.
+          description: A list of regions in which to describe EC2 instances. By default this is region from environment
+              environment variable or region from boto configuration or all regions except us-gov-west-1 and cn-north-1.
+          env:
+              - name: AWS_DEFAULT_REGION
```

Comment:
Thanks for the pull request @tbouvet. I think the alternative PR linked is a little better since a `describe_regions` call is made as a first attempt which will retrieve a complete list of available regions, whereas `get_available_regions` only collects the local list of regions, which could be quite out of date depending on a user's boto3 version.

However, if you don't mind simplifying this PR just to add environment variables, I think that would be a nice thing to have.  Modules use `AWS_REGION` and `EC2_REGION`, so it would be good to add those for consistency. You'll also need to add a minor_change documentation fragement https://docs.ansible.com/ansible/latest/reference_appendices/release_and_maintenance.html#creating-new-fragments.

Reply:
I would like to see this plugin use the env var AWS_REGION!

Reply:
perhaps both AWS_REGION and AWS_DEFAULT_REGION?

Reply:
The vast majority of ansible modules use `AWS_REGION`, though aws-cli now seems to be recommending `AWS_DEFAULT_REGION`. You could take the approach used in `module_utils/ec2.py` to cover all three options: https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/ec2.py#L252-L258